### ISSUE TITLE
"conda update conda" breaks conda! 

### ISSUE THREAD

**Issue by leogama** [conda/conda#3071]
Please, help! I've tried to simply update conda, but it failed miserably...
There were bad permissions on some files and directories and then the exception below was reported. After that, the conda executable disappeared from `/usr/local/conda/bin` (my installation path).

Python still works and I can even import the conda module normally, but cannot call any conda command (including `conda info`, however you can see the versions in the command output). I'm on Ubuntu 14.04.4 LTS. Thank you!

PS: I know it probably was my mistake (could have used sudo), but I think it should be recoverable in some way.

```
$ conda update conda
Fetching package metadata: ......
.Solving package specifications: ...........

Package plan for installation in environment /usr/local/conda:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    sqlite-3.13.0              |                0         4.0 MB
    xz-5.2.2                   |                0         644 KB
    python-3.5.2               |                0        17.2 MB
    conda-env-2.5.1            |           py35_0          27 KB
    ruamel_yaml-0.11.7         |           py35_0         378 KB
    conda-4.1.8                |           py35_0         204 KB
    ------------------------------------------------------------
                                           Total:        22.4 MB

The following NEW packages will be INSTALLED:

    ruamel_yaml: 0.11.7-py35_0

The following packages will be UPDATED:

    conda:       4.0.2-py35_0 --> 4.1.8-py35_0 
    conda-env:   2.4.5-py35_0 --> 2.5.1-py35_0 
    python:      3.5.1-0      --> 3.5.2-0      
    sqlite:      3.9.2-0      --> 3.13.0-0     
    xz:          5.0.5-1      --> 5.2.2-0      

Proceed ([y]/n)? 

Fetching packages ...
sqlite-3.13.0- 100% |##################################################| Time: 0:00:01   2.44 MB/s
xz-5.2.2-0.tar 100% |##################################################| Time: 0:00:00 682.48 kB/s
python-3.5.2-0 100% |##################################################| Time: 0:00:02   6.39 MB/s
conda-env-2.5. 100% |##################################################| Time: 0:00:00 175.05 kB/s
ruamel_yaml-0. 100% |##################################################| Time: 0:00:00 552.52 kB/s
conda-4.1.8-py 100% |##################################################| Time: 0:00:00 329.52 kB/s
Extracting packages ...
[      COMPLETE      ]|#####################################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|#####################################################################| 100%
Linking packages ...
An unexpected error has occurred, please consider sending the#################              |  66%
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues


Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/usr/local/conda/bin/conda", line 6, in <module>
  File "/usr/local/conda/lib/python3.5/site-packages/conda/cli/main.py", line 139, in main
    args_func(args, p)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/cli/main.py", line 146, in args_func
    args.func(args, p)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/cli/main_update.py", line 48, in execute
    install.install(args, parser, 'update')
  File "/usr/local/conda/lib/python3.5/site-packages/conda/cli/install.py", line 424, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/plan.py", line 539, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/instructions.py", line 149, in execute_instructions
    cmd(state, arg)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/instructions.py", line 92, in LINK_CMD
    link(state['prefix'], arg, index=state['index'])
  File "/usr/local/conda/lib/python3.5/site-packages/conda/instructions.py", line 88, in link
    install.link(pkgs_dir, prefix, dist, lt, index=index)
  File "/usr/local/conda/lib/python3.5/site-packages/conda/install.py", line 631, in link
    os.makedirs(dst_dir)
  File "/usr/local/conda/lib/python3.5/os.py", line 241, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/usr/local/conda/lib/python3.5/site-packages/ruamel_yaml'
```

**Comment by leogama** [comment]
PS: I know it probably was may mistake, but I think it should be recoverable in some way.

**Comment by leogama** [comment]
Clicked in "Close and comment" accidentally...

**Comment by kalefranz** [comment]
@HugoTian, this is a permissions issue that's a *very* high priority to solve. Your current project should help a lot. There are a bunch of bug reports just like this going back years. Let's figure out exactly how all this is happening.

**Comment by kalefranz** [comment]
CC @ijstokes

**Comment by leogama** [comment]
Does the `ruamel_yaml` package is a dependency of conda? Because if it is not and conda was installed before it (and before any other potential fail), it would be easy to fix the installation after.

BTW, is there any quick and dirty fix that I can use to keep my current installation? 🤔

**Comment by kalefranz** [comment]
Yeah, here's an example. Customize for your situation.

    curl http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o ~/miniconda3.sh
    bash ~/miniconda3.sh -bfp /usr/local/conda

**Comment by kalefranz** [comment]
The root of the problem has something to do with permissions. (FWIW, I'm still completely perplexed by how posix permissions ever allow this to happen.)

For now, if you install into `/usr/local/conda` as `root`, than *always* use `sudo`.  Or, install into `/usr/local/conda` as root, and then `chown -R`, and *never* uses sudo again.

**Comment by leogama** [comment]
As long as I can remember, I've done exactly what you describes as the second option:

> Or, install into `/usr/local/conda` as `root`, and then `chown -R`, and _never_ uses `sudo` again.

I can't check this because I've called `chown -R` in the installation directory after the failed update (but before discovering that conda itself was broken).

**Comment by leogama** [comment]
The fix suggested seems to have worked. Thank you so much!

**Comment by HugoTian** [comment]
@leogama  May I know how you install conda ? I try to reproduce the error and I get following error messages

Error: Missing write permissions in: /usr/local/conda
#
# You don't appear to have the necessary permissions to update packages
# into the install area '/usr/local/conda'.
# However you can clone this environment into your home directory and
# then make changes to it.
# This may be done using the command:
#
# $ conda create -n my_root --clone=/usr/local/conda

**Comment by stanep** [comment]
Same happens to last night, lost "conda" I'll try tonight 
curl http://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o ~/miniconda3.sh
bash ~/miniconda3.sh -bfp /usr/local/conda

**Comment by kalefranz** [comment]
Fixed with conda 4.3.
